### ISSUE TITLE
Is optimizer.zero_grad() intentionally omitted in training loop?

### ISSUE THREAD

**Issue by salmooonaa** [wz0919/waypoint-predictor#4]
Hi, thanks for sharing this great project!
I noticed that in your training loop, there doesn’t seem to be a call to `optimizer.zero_grad()`. [code](https://github.com/wz0919/waypoint-predictor/blob/main/waypoint_predictor.py/#L153-L155)
Is this intentional? I was wondering if gradients are being cleared elsewhere, or if this might be an oversight.

Thanks!
